plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergLocalConcurrentWrites.java

```diff
@@ -361,6 +362,13 @@ void testConcurrentTruncateAndInserts()
     @RepeatedTest(3)
```

Comment:
The tests seem to be quite complex compared to the actual code change. Not sure sure if they covering all scenarios like change in partitioning spec, unpartitioned etc.
Is it possible to make to instead make unit tests on `extractTupleDomainsFromCommitTasks` which just assert on expected domain from various scenarios ? I don't think you need to run actual queries with concurrency to test the logic here.
You could look at `io.trino.plugin.hive.TestHivePageSink` as a reference for how we can operate on a connector page sink directly, that can be used to create `CommitTaskData` for different scenarios.

Reply:
Added unit tests `TestFileBasedConflictDetection` please take a look